source/dotnet/Library/AdaptiveCards/AdaptiveInlinesConverter.cs

```diff
@@ -35,7 +41,22 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                 else
                 {
                     JObject jobj = (JObject)obj;
-                    arrayList.Add((AdaptiveInline)jobj.ToObject(typeof(AdaptiveTextRun)));
+                    if (jobj.Value<string>("type") != AdaptiveTextRun.TypeName)
+                    {
+                        throw new AdaptiveSerializationException($"Property 'type' must be '{AdaptiveTextRun.TypeName}'");
+                    }
+
+                    if (ParseContext == null)
+                    {
+                        ParseContext = new ParseContext();
+                    }
+
+                    var adaptiveInline = JsonConvert.DeserializeObject<AdaptiveTextRun>(jobj.ToString(), new JsonSerializerSettings
+                    {
+                        ContractResolver = new WarningLoggingContractResolver(new AdaptiveCardParseResult(), ParseContext),
+                        Converters = { new StrictIntConverter() }
+                    });
```

Comment:
i think this one is missing a latest change

Reply:
@nesalang -- can you elaborate? it does look like @shalinijoshi19 's revert of the serialization stuff impacted this code...

Reply:
LN27-29 has a the latest, and LN 49-55 had one older commit. I checked-in the fix via iteration 20

---
In reply to: [342306159](https://github.com/microsoft/AdaptiveCards/pull/3565#discussion_r342306159) [](ancestors = 342306159)

Reply:
awesome! thanks!